internal/pkg/v2/infrastructure/redis/event.go

```diff
@@ -0,0 +1,68 @@
+//
```

Comment:
@andresrinivasan , needs to review this code.

Reply:
I'm curious why the move from storing the object as a blob to storing the object broken down in a hash. The current query interface won't be able to take advantage of this and when an object is returned, it's the entire object, not a subset of fields.

Reply:
The reason that we used hashes is that it looks like a good way to represent a data model(a Golang Struct data type) as a Redis data structure. The `AddFlat` and `ScanStruct` methods in `github.com/gomodule/redigo/redis` library is easy to deal with structs for the `HSET` and `HGET` redis commands so event if we don't need to get a specific key in the object, we can still easily to convert between Hashes and models. 
Please let us if any other suggestion or concern, thanks.

Reply:
Agreed that the Hash data structure can be used to mirror the Go struct. From the perspective of the EdgeX API, we mostly query on indexes that aren't well represented by a Hash and we always return the entire struct. That's why the current implementation denormalizes the data into multiple indexes that all point to the same blob that will be returned.

Reply:
@lindseysimple , @cloudxxx8 , I discussed the merits between using `Hash` and using `Blob` with @andresrinivasan today.

The use of `hash`, while functional,  is not required for our implementation and is slightly less efficient. One benefit of staying with use of `Blob`, as done for V1, is we can easily review the new V2 code using the V1 code as the standard, without needing @andresrinivasan to participate, even when there are slight differences in the V2 implementation's use of `Blob`.

Thoughts?

Reply:
@lenny-intel Lenny, Thanks for the update. Discussed with @cloudxxx8 today and we agree that save as `Blob` will be more suitable for the case here since we return the entire object in the API response instead of a single field. 

Already updated the infrastructure layer functions to save as blob as in v1.